subsys/bluetooth/host/classic/br.c

```diff
@@ -1160,6 +1160,33 @@ int bt_br_set_discoverable(bool enable)
 	}
 }
 
+int bt_br_set_visibility(bool disc_mode, bool conn_mode)
+{
+	if (disc_mode && conn_mode) {
```

Comment:
cloud we Simple it with as below?
uint8 flag = 0 ;

if(disc_mode) {
  flag |= XX;
  } else {
   flag~= XX;
   }
  
  if(conn_mode) {
  flag |= XX;
  }else {
   flag~= XX;
   }
  
  write_scan_enable（flag）